datafusion/common/src/scalar.rs

```diff
@@ -2039,7 +2039,11 @@ impl ScalarValue {
         }
     }
 
-    /// Retrieve ScalarValue for each row in `array`
+    /// Retrieve `ScalarValue` for each row in `array`
+    ///
+    /// Convert `ListArray` to `Vec<Vec<ScalarValue>>`, first `Vec` is for rows, second `Vec` is for elements in the list
```

Comment:
👍 It might also help to explain why we need two different signatures. It took me a while to grok things too

```suggestion
    /// Convert `ListArray` into a 2 dimensional to `Vec<Vec<ScalarValue>>`, first `Vec` is for rows, 
    /// second `Vec` is for elements in the list.
    ///
    /// See [`Self::convert_non_list_array_to_scalars`] for converting non Lists
    ///
    /// This method is an optimization to unwrap nested ListArrays to nested Rust structures without
    /// converting them twice
```